frontend/src/components/Header.tsx

```diff
@@ -0,0 +1,78 @@
+import React, {useState} from 'react';
+import {getOrCreateUser} from "../services/api.ts";
+import {RegisterUserRequest} from "../data/RegisterUserRequest.ts";
+import './Header.css';
+import {UserDto} from "../data/UserDto.ts";
+
+const Header: React.FC = () => {
+    const [inputValue, setInputValue] = useState<string>('');
+    const [user, setUser] = useState<UserDto | null>(null);
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    const validateUsername = (name: string): boolean => {
+        const usernamePattern = /^[A-Za-z0-9]+$/; // Nur Buchstaben und Zahlen erlaubt
```

Comment:
Code sollte nach aller Möglichkeit Selbstdokumentierend sein. Wenn man Kommentare verwendet, dann eher um zu erklären warum etwas gemacht wird und nicht was gemacht wird. In diesem Fall würde ich also sowas machen wie `const usernamePattern = allowLettersAndNumbers()` oder direkt `allowLettersAndNumbersPattern = /.../`